Ch1 console games

PavithraY22 opened the pull request:
## Location

Specify chapters and sections here.

- Chapter(s): 1
- Section(s): Console games

## Changes

- Created 3 new games for ch1

## Notes

- Add any notes here for the Curriculum Development team


**Add all relevant labels to this PR (e.g. if it's about a practice problem, add the practice problem label)**

PavithraY22 committed:
Create guess the password

PavithraY22 committed:
Create Tower of Courage

PavithraY22 committed:
Create Blackjack

BenVN123 committed:
Rename Blackjack to Blackjack.py

lint-action committed:
Fix code style issues with Black

BenVN123 committed:
Rename Tower of Courage to TowerOfCourage.py

BenVN123 committed:
Rename guess the password to guessthepassword.py

BenVN123 committed:
Fix styling errors

BenVN123 committed:
Fix styling errors

lint-action committed:
Fix code style issues with Black

BenVN123 committed:
Fix styling errors

chrehall68 commented on guessthepassword.py:
```diff
@@ -0,0 +1,20 @@
+# you set and see how many tries it takes for that user to guess correctly
+
+# start with assigning the password to some variable
+pas = str("password")
```
you don't actually need the str() since "password" is already a string

chrehall68 commented on TowerOfCourage.py:
```diff
@@ -0,0 +1,39 @@
+# Story + Directions: This Summer, an amusement
```
Not exactly a game.

chrehall68 reviewed (changes requested):
It mostly looks good. Keep in mind that, since students will have finished our intermediate course that already deals with functions and classes, you can make the problems a bit harder. I recommend that you try to come up with another example to replace TowerofCourage.py

PavithraY22 committed:
Update guessthepassword.py

PavithraY22 committed:
Update TowerOfCourage.py

lint-action committed:
Fix code style issues with Black

PavithraY22 commented:
> It mostly looks good. Keep in mind that, since students will have finished our intermediate course that already deals with functions and classes, you can make the problems a bit harder. I recommend that you try to come up with another example to replace TowerofCourage.py

I am not too familiar with classes however I can incorporate them in game if you let me know how to create a certain game with classes. I created another game with functions called "Hangman" so please look into that.

PavithraY22 committed:
Rename TowerOfCourage.py to hangman.py

BenVN123 committed:
Fix styling errors

chrehall68 committed:
Fix a couple bugs, clarify instructions

One bug was found in "if guess != wordSet". This is always True since guess is a string and wordSet is a set. Since we only want to decrement guesses when the user guesses wrong,  I changed it to "if guess not in wordSet". Also, I defined score by setting it to 0 (before, it was undefined and would error). I made the hangman function take as a parameter the endword. I clarified the instructions to fit. Also, fixed a bug where you would 'lose' and 'win' at the same time if you guessed the last letter of the word on your last guess.

chrehall68 committed:
Move ch.1 games to new games/chapter1 folder

lint-action committed:
Fix code style issues with Black

chrehall68 committed:
Use f"{guesses}"

chrehall68 committed:
Merge branch 'CH1-console-games' of https://github.com/code4tomorrow/python into CH1-console-games

chrehall68 committed:
Replace 'guesses' with 'lives'

chrehall68 reviewed (approved):
LGTM now.

BenVN123 committed:
Add newline

BenVN123 committed:
Rename games/chapter1/Blackjack.py to games/chapter1/examples/blackjack.py

BenVN123 committed:
Rename games/chapter1/guessthepassword.py to games/chapter1/examples/guessthepassword.py

BenVN123 committed:
Rename games/chapter1/hangman.py to games/chapter1/examples/hangman.py

BenVN123 reviewed (approved):
Looks good.
